core/src/main/java/io/questdb/cairo/vm/MemoryCMARWImpl.java

```diff
@@ -73,7 +73,9 @@ public void close(boolean truncate, byte truncateMode) {
             if (truncate) {
                 long appendOffset = getAppendOffset();
                 truncateSize = truncateMode == Vm.TRUNCATE_TO_PAGE ? Files.ceilPageSize(appendOffset) : appendOffset;
-                long sz = Math.min(size, truncateSize);
+                // If this is lazy close of unused memory the underlying file can be already truncated using another fd
+                // and memset can lead to SIGBUS on Linux. Check the phisical file length before trying to memset to the map
+                long sz = Math.min(ff.length(fd), Math.min(size, truncateSize));
```

Comment:
Unlike other ff calls used here, `ff.length()` may actually throw. We may want to handle the exception here (or introduce a non-throwing version of `length()`) and log it.

Reply:
done, added a catch here